Automattic/wp-calypso
# Full Site Editing: remove template code duplication from themes

#### Changes proposed in this Pull Request

To avoid duplicating this file across all FSE themes I'm moving it to the FSE plugin instead and providing hooking to theme activation there. This is now viable since we are no longer hardcoding the templates in themes, but fetching them via API.

Fixes https://github.com/Automattic/wp-calypso/issues/35191
Depends on https://github.com/Automattic/themes/pull/1216

#### Testing instructions

1. Apply https://github.com/Automattic/themes/pull/1216 to update your Modern Business theme.
2. Delete previous Templates entries. You can access them at `wp-admin/edit.php?post_type=wp_template`.
4. Map this theme to your local Gutenberg Docker dev environment, in a similar fashion you do for FSE plugin: `{themes_repo_path}/themes/modern-business:/var/www/html/wp-content/themes/modern-business`
5. Delete `modern-business-fse-template-data-v1` option if it was set on your site. This probably won't be the case for the first run, since I've bumped its version.
6. Activate Modern Business theme to trigger the activation hook.
7. Make sure that Templates are populated with correct content. (`wp-admin/edit.php?post_type=wp_template`)
8. Make sure that everything still works as before.

To test the flow where theme is activated prior to FSE plugin:

1. Delete previous Templates entries. You can access them at `wp-admin/edit.php?post_type=wp_template`.
2. Deactivate Full Site Editing plugin.
3. Delete `modern-business-fse-template-data-v1` option if it was set on your site (or just bump it in the code to `v123` for testing purposes).
4. Make sure that you are currently running an FSE supported theme (only Modern Business atm).
5. Activate the Full Site Editing plugin.
6. Make sure that Templates are populated with correct content. (`wp-admin/edit.php?post_type=wp_template`)
7. This population should occur if plugin is activated while a theme that doesn't support FSE is active (e.g. Twenty Nineteen).
